### ISSUE TITLE
Panic when using websocket and backend gets removed

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#380]
When I have a backend that does websocket and a client connected to it, if I stop the backend, a worker panics:
```
2018-03-05T00:10:46Z 68253955278537 19087 WRK-00 DEBUG  sozu_lib::network::protocol::pipe       58c522d3-deb5-47cb-b187-2cea58fd072a    unknown         FRONT [Some(Token(4))]: read 0 bytes
2018-03-05T00:10:46Z 68253955309707 19087 WRK-00 TRACE  sozu_lib::network::http front readable  interpreting client order Continue
2018-03-05T00:10:46Z 68253955333878 19087 WRK-00 TRACE  sozu_lib::network::protocol::pipe       pipe back_readable                                                                          
2018-03-05T00:10:46Z 68253955365048 19087 WRK-00 DEBUG  sozu_lib::network::protocol::pipe       58c522d3-deb5-47cb-b187-2cea58fd072a    unknown         BACK  [4<-5]: read 0 bytes
2018-03-05T00:10:46Z 68253955452348 19087 WRK-00 DEBUG  sozu_lib::network::http         PROXY [4 -> 5] CLOSED BACKEND                                                                                                                                         
thread 'main' panicked at 'lib/src/network/http.rs:265 sozu_lib::network::http this should not have panicked:                          
tried to unwrap `None` in:                                                                                                                                                                                                          
unwrap_msg!(self.http())', lib/src/util.rs:18:18    
```

This is here: https://github.com/sozu-proxy/sozu/blob/863bfb5b849d84319d452e2c85b50e369420f8a0/lib/src/network/http.rs#L265

I'm using this websocket example (from https://github.com/websockets/ws#expressjs-example)

server.js
```javascript
const express = require('express');
const http = require('http');
const WebSocket = require('ws');

const app = express();

app.use(function (req, res) {
  res.send({ msg: "hello" });
});

const server = http.createServer(app);
const wss = new WebSocket.Server({ server });

wss.on('connection', function connection(ws, req) {
  ws.on('message', function incoming(message) {
    console.log('received: %s', message);
    ws.send(Date.now());
  });
});

server.listen(1026, function listening() {
  console.log('Listening on %d', server.address().port);
});
```

client.js:
```javascript
const WebSocket = require('ws');

const ws = new WebSocket('ws://lolcatho.st:8080', {
  origin: 'http://lolcatho.st'
});

ws.on('open', function open() {
  console.log('connected');
  ws.send(Date.now());
});

ws.on('close', function close() {
  console.log('disconnected');
});

ws.on('message', function incoming(data) {
  console.log(`Roundtrip time: ${Date.now() - data} ms`);

  setTimeout(function timeout() {
    ws.send(Date.now());
  }, 500);
});
```

Start the server and client, then kill the server. The panic should occur right away

**Comment by Geal** [comment]
this should be fixed by 2ceae33

**Comment by BlackYoup** [comment]
Indeed we can :)
